pwa-devdocs/src/tutorials/pwa-studio-fundamentals/add-a-static-route/index.md

```diff
@@ -0,0 +1,284 @@
+
+```diff
+  const Search = lazy(() => import('@magento/venia-ui/lib/RootComponents/Search'));
++ const Foo = lazy(() => import('../Foo'));
+```
+
+### Add custom route
+
+Add a new route entry inside the Switch component and assign the Foo component for rendering that route:
+
+```diff
+  <Suspense fallback={fullPageLoadingIndicator}>
+      <Switch>
++         <Route exact path="/foo">
++           <Foo />
++         </Route>
+          <Route exact path="/search.html">
+              <Search />
+          </Route>
+          <Route exact path="/create-account">
+              <CreateAccountPage />
+          </Route>
+          <Route exact path="/cart">
+              <CartPage />
+          </Route>
+          <Route exact path="/checkout">
+              <CheckoutPage />
+          </Route>
+          <Route>
+              <MagentoRoute />
+          </Route>
+      </Switch>
+  </Suspense>
+```
+
+### Create Routes component's `index.js` file
+
+Create an `index.js` file for your custom Routes component to expose your `routes.js` module using the directory name.
+
+```js
+/* src/components/Routes/index.js */
+
+export {default} from './routes';
+```
+
+## Override the App component
+
+The App component is the project's main application entry point.
+The steps for overriding this component is the same as the ones for overriding the Routes component.
+
+### Create App directory
+
+Create an **App** directory to hold your project's copy of Venia's App component code.
+
+```sh
+mkdir -p src/components/App
+```
+
+### Copy over App component's modules
+
+The `app.js` file in Venia's App component imports and uses the Routes component, so copy this file into your project.
+
+```sh
+cp node_modules/@magento/venia-ui/lib/components/App/app.js src/components/App
+```
+
+If you look at the [`index.js` file for Venia's App component][], its default export is not `app.js`.
+The default export for this component is `container.js`, which is a container for the `app.js` module, so copy the `container.js` file into your project.
+
+```sh
+cp node_modules/@magento/venia-ui/lib/components/App/container.js src/components/App
+```
+
+### Update App modules' imports
+
+Open the `app.js` file and update the relative import statements to use components from the Venia UI package.
+Skip updating the Routes import to use this project's version of the Routes component.
+
+```diff
+- import { HeadProvider, Title } from '../Head';
+- import Main from '../Main';
+- import Mask from '../Mask';
+- import MiniCart from '../MiniCart';
+- import Navigation from '../Navigation';
++ import { HeadProvider, Title } from '@magento/venia-ui/lib/components/Head';
++ import Main from '@magento/venia-ui/lib/components/Main';
++ import Mask from '@magento/venia-ui/lib/components/Mask';
++ import MiniCart from '@magento/venia-ui/lib/components/MiniCart';
++ import Navigation from '@magento/venia-ui/lib/components/Navigation';
```

Comment:
Just a thought as I read all of these: should we be doing this in Venia already?